pkg/config/piped_test.go

```diff
@@ -1369,3 +1386,49 @@ func TestFindPlatformProvidersByLabel(t *testing.T) {
 		})
 	}
 }
+
+func TestPipeGitValidate(t *testing.T) {
+	t.Parallel()
+	testcases := []struct {
+		git           PipedGit
+		err error
+	}{
+		{
+			git: PipedGit{
+				SSHKeyData: "sshkey1",
+				PersonalAccessToken: PipedGitPersonalAccessToken{
+					UserName:  "UserName",
+					UserToken: "UserToken",
+				},
+			},
+			err: errors.New("cannot configure both sshKeyData or sshKeyFile and personalAccessToken"),
+		},
+		{
+			git: PipedGit{
+				SSHKeyData: "sshkey2",
+			},
+			err: nil,
+		},
+		{
+			git: PipedGit{
+				PersonalAccessToken: PipedGitPersonalAccessToken{
+					UserName:  "UserName",
+					UserToken: "UserToken",
+				},
+			},
+			err: nil,
+		},
+		{
+			git: PipedGit{ },
+			err: nil,
+		},
+	}
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.git.SSHKeyData, func(t *testing.T) {
+			t.Parallel()
+			err := tc.git.Validate()
+			assert.Equal(t, tc.err, err)
+		})
+	}
+}
```

Comment:
IMHO, the test above misses some cases.
Can you copy and paste the test bellow, fix `TODO`s, and run it?

```go
func TestPipeGitValidate(t *testing.T) {
	t.Parallel()
	testcases := []struct {
		name string
		git  PipedGit
		err  error
	}{
		{
			name: "both SSH and PAT are valid",
			git: PipedGit{
				SSHKeyData: "sshkey1",
				PersonalAccessToken: PipedGitPersonalAccessToken{
					UserName:  "UserName",
					UserToken: "UserToken",
				},
			},
			err: errors.New("cannot configure both sshKeyData or sshKeyFile and personalAccessToken"),
		},
		{
			name: "Both SSH and PAT is not valid",
			git: PipedGit{
				SSHKeyFile: "sshkeyfile",
				SSHKeyData: "sshkeydata",
				PersonalAccessToken: PipedGitPersonalAccessToken{
					UserName:  "",
					UserToken: "UserToken",
				},
			},
			// TODO: should return error
		},
		{
			name: "SSH key data is not empty",
			git: PipedGit{
				SSHKeyData: "sshkey2",
			},
			err: nil,
		},
		{
			name: "SSH key file is not empty",
			git: PipedGit{
				SSHKeyFile: "sshkey2",
			},
			err: nil,
		},
		{
			name: "Both SSH file and data is not empty",
			git: PipedGit{
				SSHKeyData: "sshkeydata",
				SSHKeyFile: "sshkeyfile",
			},
			// TODO: should return error
		},
		{
			name: "PAT is valid",
			git: PipedGit{
				PersonalAccessToken: PipedGitPersonalAccessToken{
					UserName:  "UserName",
					UserToken: "UserToken",
				},
			},
			err: nil,
		},
		{
			name: "PAT username is empty",
			git: PipedGit{
				PersonalAccessToken: PipedGitPersonalAccessToken{
					UserName:  "UserName",
					UserToken: "",
				},
			},
			// TODO: should return error
		},
		{
			name: "PAT token is empty",
			git: PipedGit{
				PersonalAccessToken: PipedGitPersonalAccessToken{
					UserName:  "",
					UserToken: "UserToken",
				},
			},
			// TODO: should return error
		},
		{
			name: "Git config is empty",
			git:  PipedGit{},
			err:  nil,
		},
	}
	for _, tc := range testcases {
		tc := tc
		t.Run(tc.git.SSHKeyData, func(t *testing.T) {
			t.Parallel()
			err := tc.git.Validate()
			assert.Equal(t, tc.err, err)
		})
	}
}
```